pkg/option/config.go

```diff
@@ -2450,6 +2460,26 @@ func (c *DaemonConfig) Populate() {
 	c.DisableCNPStatusUpdates = viper.GetBool(DisableCNPStatusUpdates)
 }
 
+func (c *DaemonConfig) populateDevices() {
+	c.Devices = viper.GetStringSlice(Devices)
+	if device := viper.GetString(Device); device != "" {
+		if c.Devices == nil {
+			c.Devices = []string{device}
+		} else {
+			found := false
+			for _, d := range c.Devices {
+				if d == device {
+					found = true
+					break
+				}
+			}
+			if !found {
+				c.Devices = append(c.Devices, device)
+			}
+		}
```

Comment:
Nit: non-blocking comment--to keep the code less indented, you could return out of the `if` statement, and convert the `else` block to simply fallthrough